x-pack/plugins/reporting/server/features.ts

```diff
@@ -37,6 +38,7 @@ export function registerFeatures({
         defaultMessage: 'Reporting',
       }),
       category: DEFAULT_APP_CATEGORIES.management,
+      scope: [KibanaFeatureScope.Spaces, KibanaFeatureScope.Security],
```

Comment:
> seems that all features without the Spaces scope will be treated as "disabled"

Correct. We agreed to not add any `space` scope for any future features and treat the current set of features as locked. As for the serverless only features, I think it makes sense to remove `space` scope for them, but how do we distinguish which one is serverless only right now?

Reply:
Checked that out, for now it's only the `reporting` feature. I can create a separate test for serverless to check this.

Reply:
Yeah, I think it's only `reporting` for now.

> As for the serverless only features, I think it makes sense to remove space scope for them.

It makes sense to me, but the new logic we have in `capabilities_switcher` might be problematic. If I’m reading the code correctly, serverless features will be treated as `disabledFeatures` in this case, meaning all the nav links, catalogue entries, and management item capabilities exposed by them will be disabled as well, which isn’t what we want, I guess. If I understand the intention correctly, we only want features without `space` scope to not be displayed in the feature toggles list, and potentially end up in `space.disabledFeatures`. So, do we need the change in `capabilities_switcher`, or would filtering these features out on the feature toggles screen and rejecting them in `disabledSpaces` when a space is created or updated be enough?

Reply:
> we only want features without space scope to not be displayed in the feature toggles list, and potentially end up in space.disabledFeatures
yeah, as for the change in `capabilities_switcher` I think we do, because this is the `root` point for it and after we set up capabilities it gets passed down to all other components. Though, I might also misunderstand smth.

Reply:
@elena-shostak and I discussed this over Slack, and here's the summary:

Basically, the lack of `spaces` scope serves as an implicit alternative to including a feature in the `space.disabledFeatures` list. In this sense, treating them similarly in `capabilities_switcher` makes total sense.

What confused me is that the `spaces` scope only makes sense when feature toggles - and the corresponding functionality `spaces` scope replicates (`space.disabledFeatures`) - exist. However, in serverless, there are no feature toggles, but the lack of `spaces` scope still affects functionality, effectively disabling it in `capabilities_switcher`.

We probably need to discuss this further as the functionality/project evolves, but for now, we agreed that it likely makes sense to add `spaces` scope to all serverless-only features to ensure they aren't accidentally disabled, along with a code comment explaining why it's needed.

Reply:
@azasypkin, @elena-shostak:

Great catch, and thanks for the discussion. My expectation is that the changes in this PR will not impact any capability-switching logic. The Space `scope` option only influences whether or not the feature is made available to the feature visibility toggles. If anything, we can consider this to have the opposite effect: If a feature does not have a Space `scope`, then it cannot be considered for exclusion via `disabledFeatures`. It can, however, be considered for exclusion based on the selected solution view.

This PR serves as an intermediate step towards removing the Spaces Visibility toggles altogether. The purpose is to prevent additional features from appearing in the visibility toggle list, so that we have more control over the set of visible features through mechanisms such as the solution view.

Reply:
After clarification over Slack, reverted change to `capabilities_switcher` in e45363c11f9643bda4f979c40f1a0029d7307e7a